### DISCUSSION TITLE
m

### DISCUSSION THREAD

**Discussion by babizhu** [socketio/socket.io#4818] [Q&A] [answered]
When I send a connect request, there will be three http requests, and the server will receive two successful connection messages. 
Is this correct?

![image](https://github.com/socketio/socket.io/assets/902091/79720d58-d018-4a82-9f68-c8bdc3dabb14)

**client :**
const opts = {
  query: {
    loginUserNum: 5,
  }
};
let socket = io('http://localhost:9099',opts)
socket.connect();

=======================================================

front : "socket.io-client": "^4.7.2",
back : <dependency>
            <groupId>com.corundumstudio.socketio</groupId>
            <artifactId>netty-socketio</artifactId>
            <version>2.0.3</version>
        </dependency>

**Comment by darrachequesne** [comment, marked as answer]
Hi! The server will receive one single "connection" event, as all three HTTP requests belong to the same Socket.IO connection.

More on that here: https://socket.io/docs/v4/how-it-works/

**Reply by babizhu** [reply]
thank you
